Index: c-decl.c
===================================================================
RCS file: /cvs/gcc/gcc/gcc/c-decl.c,v
retrieving revision 1.630.6.18
diff -u -p -r1.630.6.18 c-decl.c
--- gcc/c-decl.c	9 Sep 2005 00:51:44 -0000	1.630.6.18
+++ gcc/c-decl.c	16 Sep 2005 18:48:11 -0000
@@ -1302,10 +1302,10 @@ diagnose_mismatched_decls (tree newdecl,
 		  return false;
 		}
 	      /* If both decls have not extern inline, reject the new decl.  */
-	      if (!DECL_DECLARED_INLINE_P (olddecl)
-		  && !DECL_EXTERNAL (olddecl)
-		  && !DECL_DECLARED_INLINE_P (newdecl)
-		  && !DECL_EXTERNAL (newdecl))
+	      if (!(DECL_DECLARED_INLINE_P (olddecl)
+		    && DECL_EXTERNAL (olddecl))
+		  && !(DECL_DECLARED_INLINE_P (newdecl)
+		       && DECL_EXTERNAL (newdecl)))
 		{
 		  error ("%Jredefinition of %qD", newdecl, newdecl);
 		  locate_old_decl (olddecl, error);
