Backport from gcc-3.4.5

--- branches/gcc-3_4-branch/gcc/cp/pt.c	2005/09/02 11:20:21	103770
+++ branches/gcc-3_4-branch/gcc/cp/pt.c	2005/09/03 19:53:01	103815
@@ -4810,6 +4810,7 @@
 	   || TREE_CODE (t) == TEMPLATE_PARM_INDEX
 	   || TREE_CODE (t) == OVERLOAD
 	   || TREE_CODE (t) == BASELINK
+	   || TREE_CODE (t) == IDENTIFIER_NODE
 	   || TREE_CODE_CLASS (TREE_CODE (t)) == 'c')
     dependent_p = (type_dependent_expression_p (t)
 		   || value_dependent_expression_p (t));
@@ -12028,7 +12029,8 @@
 	    return true;
 	  expression = TREE_OPERAND (expression, 0);
 	}
-      if (TREE_CODE (expression) == OVERLOAD)
+      if (TREE_CODE (expression) == OVERLOAD
+	  || TREE_CODE (expression) == FUNCTION_DECL)
 	{
 	  while (expression)
 	    {
