*** gcc/cp/decl2.c 2003-01-16 09:03:26.000000000 +0100
--- gcc/cp/decl2.c 2003-01-16 17:18:11.000000000 +0100
*************** arg_assoc (struct arg_lookup *k, tree n)
*** 4037,4042 ****
--- 4037,4045 ----
    if (TYPE_P (n))
      return arg_assoc_type (k, n);

+   if (TREE_CODE (n) == NEW_EXPR)
+     return arg_assoc_type (k, TREE_OPERAND (n, 1));
+
    if (! type_unknown_p (n))
      return arg_assoc_type (k, TREE_TYPE (n));

